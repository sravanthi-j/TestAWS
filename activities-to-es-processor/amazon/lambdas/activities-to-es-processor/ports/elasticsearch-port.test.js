
const EsPort = require("./elasticsearch-port").ElasticsearchPort;
const EventEmitter = require('events');
const expect = require('expect');

class ResponseEmitter extends EventEmitter {

    constructor(message, statusCode)
    {
        super();
        this.statusCode = statusCode;
        this.message = message;
    }

    emitMessage(){
        this.emit('data', this.message);
        this.emit('end');
    }
    end() {
 
    }
     
}


class MockHttps
{   
    constructor(emitter)
    {
        this.emitter = emitter;
    }

    request(requestParam, callback){
 
        callback(this.emitter);
        this.emitter.emitMessage();
        this.emitter.removeAllListeners();
    }

}
 


describe('Peforms request', () => {

 test('With one fail and one success', async () => {

    var esPort = new EsPort(
        new MockHttps(
            new ResponseEmitter(JSON.stringify({
                items: [
                    {index: {status: 500}},
                    {index: {status: 200}}
                ]
            }), 200
        ))); 
        var response = await esPort.post({});
        expect(response.attemptedItems).toBe(2);
        expect(response.failedItems).toBe(1);
        expect(response.successfulItems).toBe(1);    
  });
  test('Promise rejects with error', () => {

      var esPort = new EsPort(
        new MockHttps(
            new ResponseEmitter(JSON.stringify({
                items: [
                    {index: {status: 500}}
                 ]
            }), 500
        ))); 
        return expect(esPort.post({})).rejects.toMatchSnapshot();
    
  });
  

});


