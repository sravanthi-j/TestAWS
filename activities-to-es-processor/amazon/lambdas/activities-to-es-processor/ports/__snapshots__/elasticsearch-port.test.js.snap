// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peforms request Promise rejects with error 1`] = `[Error: Error: Received 500 error message from the server]`;
